async_downloads/static/js/ws_async_downloads.js

```diff
@@ -0,0 +1,185 @@
+        download_div.attr("data-url", download.url);
+        bar.attr({
+            style: `width: ${download.percentage}%`,
+            "aria-valuenow": download.percentage,
+        });
+        if (download.complete) {
+            download_div.attr("complete", "true");
+            bar.addClass("bg-success");
+            bar.removeClass("bg-info progress-bar-animated");
+        }
+        if (download.errors) {
+            download_div.attr("complete", "true");
+            bar.addClass("bg-danger");
+            bar.removeClass("bg-success");
+            download_div.attr({
+                "data-toggle": "tooltip",
+                "data-placement": "bottom",
+                title: download.errors,
+            });
+            $("#async-downloads [data-toggle='tooltip']").tooltip();
+        }
+
+        if (!$('.download-container[complete="false"]').length) {
+            $("#async-downloads").removeClass("in-progress");
+        }
+    }
+
+    function removeDownload(data) {
+        $(`#download-${data.downloadKey}`).remove();
+        if ($("#downloads-dropdown > div").length === 0)
+            $("#downloads-dropdown").append(
+                '<a class="dropdown-item" href="">No downloads</a>'
+            );
+        if (!$('.download-container[complete="false"]').length) {
+            $("#async-downloads").removeClass("in-progress");
+        }
+    }
+
+    function startDownloadSocket() {
+        const url = new URL($("#async-downloads-script").data("url"));
+        url.protocol = url.protocol == "https:" ? "wss:" : "ws";
+        const ws = new WebSocket(url);
+
+        ws.onopen = function (e) {
+            ws.send(
+                JSON.stringify({
+                    eventType: "initializeAllDownloads",
+                    location: window.location.host,
+                })
+            );
+            // clear btn for download
+            $(document).on("click", ".download-clear", function (e) {
+                e.stopPropagation();
+                ws.send(
+                    JSON.stringify({
+                        eventType: "clearDownload",
+                        location: window.location.host,
+                        data: {
+                            filepath: $(this)
+                                .closest(".download-content")
+                                .closest(".download-container")
+                                .data("filepath"),
+                        },
+                    })
+                );
+            });
+            setInterval(calculateTimeStamps, 1000);
+        };
+
+        ws.onmessage = function (e) {
+            // Do nothing if the downloads div doesn't exist
+            if ($("#async-downloads").length === 0) return;
+            const msg = JSON.parse(e.data);
+            switch (msg.eventType) {
+                case "initializeAllDownloads":
+                    initializeAllDownloads(msg.data);
```

Comment:
Can we rename to `initAllDownloads`? Keeps it consistent with the `initDownload` function.

Reply:
757d4e4